elasticsearch/class-health-command.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Automattic\VIP\Elasticsearch;
+
+use \WP_CLI;
+use \WP_CLI\Utils;
+use \WP_Query as WP_Query;
+use \ElasticPress\Indexables as Indexables;
+use \ElasticPress\Elasticsearch as Elasticsearch;
+
+/**
+ * Helper commands to manage VIP Go Elasticsearch indexes
+ *
+ * @package Automattic\VIP\Elasticsearch
+ */
+class Health_Command extends \WPCOM_VIP_CLI_Command {
+
+	/**
+	 * @var WP_CLI\Fetchers\Plugin Plugin fetcher
+	 */
+	protected $plugin_fetcher;
+
+	public function __construct() {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		parent::__construct();
+
+		$this->plugin_fetcher = new \WP_CLI\Fetchers\Plugin();
+	}
+
+	/**
+	 * Validate DB and ES index post counts
+	 *
+	 * ## OPTIONS
+	 *
+	 *
+	 * ## EXAMPLES
+	 *     wp vip-es health validate-counts
+	 *
+	 * @subcommand validate-counts
+	 */
+	public function validate_counts( $args, $assoc_args ) {
+		// Make sure ElasticPress is installed and active
+		$this->is_elasticpress_active();
+
+		// Get indexable objects
+		$indexable = Indexables::factory()->get( 'post' );
+
+		$post_types = $indexable->get_indexable_post_types();
+
+		WP_CLI::line( sprintf( "Checking %d post types (%s)\n", count( $post_types ), implode( ',', $post_types ) ) );
+
+		$error = false;
+		$es_conn_err = false;
+
+		foreach( $post_types as $post_type ) {
+			$query_args = [ 
+				'post_type' => $post_type,
+			];
+
+			// Get total count in DB
+			$result = $indexable->query_db( $query_args );
+
+			$db_total = $result[ 'total_objects' ];
+
+			// Get total count in ES index
+			$query = new WP_Query( $query_args );
+
+			$formatted_args = $indexable->format_args( $query->query_vars, $query );
+
+			$es_result = $indexable->query_es( $formatted_args, $query->query_vars );
+			
+			if ( ! $es_result ) {
+				$es_total = 'N/A';
+				$error = true;
+				$es_conn_err = true;
+			} else {
+				$es_total = $es_result[ 'found_documents' ][ 'value' ];
+			}
+
+			$icon = "\u{2705}"; // unicode check mark
+			$diff = '';
+			if ( $db_total !== $es_total ) {
+				$error = true;
+
+				$icon = "\u{274C}"; // unicode cross mark
+
+				if ( $es_result ) {
+					$diff = sprintf( ', diff: %d', $es_total - $db_total );
+				}
+			}
+
+			WP_CLI::line( sprintf( "%s %s (db: %d, es: %s%s)", $icon, $post_type, $db_total, $es_total, $diff ) );
+		}
+
+		WP_CLI::line( '' );
+
+		if( $error ) {
+			if ( $es_conn_err ) {
+				$msg = 'cannot connect to Elasticsearch instance.';
+			} else {
+				$msg = 'found inconsistent counts for post types.';
+			}
+			WP_CLI::error( $msg );
+		}
+
+		WP_CLI::success( 'counts for public post types are all equal!' );
+	}
+
+	protected function is_elasticpress_active() {
```

Comment:
I'm not so sure about that. I wanted the check to be done every time the command is called and I also don't want to create a circular dependency between the command class and the `Elasticsearch` class. Perhaps a separate `Validator` class?

Reply:
Would `use \ElasticPress\Indexables as Indexables;` fatal if the plugin isn't loaded?

@nickdaugherty and I chatted a bit about this earlier today, and we're thinking that it makes sense for us to bundle and auto-load the ElasticPress plugin via `Automattic\VIP\Elasticsearch\Elasticsearch` (we can do this is in a future PR). 

Anyway, I think it's okay to leave as-is but probably can be moved (or removed) in the future.

Reply:
That sounds like a plan - since this PR includes the base `Elasticsearch` class that we'll need to use to load ElasticPress, let's leave it as-is in this PR then follow up by adding ElasticPress to the `elasticsearch` directory, and auto-load it when `USE_VIP_ELASTICSEARCH` is `true`.